odsclient/core.py

```diff
@@ -251,6 +256,32 @@ def get_whole_dataset(self,
 
         return result
 
+    # noinspection PyShadowingBuiltins
+    def push_dataset_realtime(self,
+                              dataset_id,         # type: str
+                              payload,            # type: List[Dict[str,Any]]
+                              push_key,           # type: str
```

Comment:
`push_key` should probably be removed here. See `get_whole_dataset` and this section of the user guide https://smarie.github.io/python-odsclient/#c-declaring-an-api-key , I guess we could rely on the same mechanisms for push, what do you think ?

Reply:
I've started working like that to conform the `api_key` logic provided by the lib. But unfortunately, it seems to me that when you are using with realtime pushes, you cannot use your API key, but for each data source you must use the `push_key` provided by ODS for this specific dataset.

Reply:
ah ok then, fair enough - let's leave it here for now. Maybe you can add some explicit comment in the docstring stating that this is not the same key (and explaining how/where to get it ?).